docs/adr/001-provisioning.md

```diff
@@ -0,0 +1,346 @@
+- Kyma Environment Broker must not contain all the details of the cluster infrastructure.
+- Kyma Infrastructure Manager's API must expose properties that:
+  - can be set in the BTP cockpit by the user
+  - are directly related to plans in the KEB
+- Kyma Infrastructure Manager's API must not expose properties that are:
+  - hardcoded in the Provisioner, or the KEB
+  - statically configured in the management-plane-config
+
+Kyma Environment Broker has the following responsibilities:  
+- Create Runtime CR containing the following data:
+    - Provider config (type, region, and secret with credentials for hyperscaler)
+    - Worker pool specification
+    - Cluster networking settings (nodes, pods, and services API ranges)
+    - OIDC settings
+    - Cluster administrators list
+    - Egress network filter settings
+    - Control Plane failure tolerance config
+- Observe the status of the CR to determine whether provisioning succeeded
+
+ Kyma Infrastructure Manager has the following responsibilities:
+- Create shoots based on:
+   - Corresponding `Runtime` CR properties
+   - Corresponding `Runtime` CR labels:
+     -  `kyma-project.io/platform-region` for determining if the cluster is located in EU 
+   - Predefined defaults for the optional properties:
+     - Kubernetes version
+     - Machine image version
+   - Predefined configuration for the following functionalities:
+     - configuring DNS extension 
+     - configuring Certificates extension
+     - providing maintenance settings (Kubernetes, and image autoupdates)
+     - creating provider specific config
+ - Upgrade and delete shoots for the corresponding `Runtime` CRs
+ - Apply the audit log configuration on the shoot resource
+ - Create cluster role bindings for administrators
+ - Generate the kubeconfig
+
+## API proposal
+
+### CR examples
+
+MInd that the Runtime CR must contain the following labels:
+```yaml
+ kyma-project.io/instance-id: instance-id
+ kyma-project.io/runtime-id: runtime-id
+ kyma-project.io/broker-plan-id: plan-id
+ kyma-project.io/broker-plan-name: plan-name
+ kyma-project.io/global-account-id: global-account-id
+ kyma-project.io/subaccount-id: subAccount-id
+ kyma-project.io/shoot-name: shoot-name
+ kyma-project.io/region: region
+ kyma-project.io/platform-region: platform-region
+ operator.kyma-project.io/kyma-name: kymaName
+```
+
+The labels are skipped in the following examples due to clarity.
+
+The example below shows the CR that must be created by the KEB to provision the AWS production cluster:
+```yaml
+apiVersion: infrastructuremanager.kyma-project.io/v1alpha1
+kind: Runtime
+metadata:
+  name: runtime-id
+  namespace: kcp-system
+spec:
+  shoot:
+    # spec.shoot.name is required
+    name: shoot-name
+    # spec.shoot.purpose is required
+    purpose: production
+    # spec.shoot.region is required
+    region: eu-central-1
+    # spec.shoot.secretBindingName is required
+    secretBindingName: "hyperscaler secret"
```

Comment:
Is KEB really managing the secrets and can provide that info on resource creation? Shouldn't the Infra manager take care of that? KEB should just translate the BTP world into our vocabular and should not deal with internal hyperscaler management in my understanding

Reply:
the problem is that selecting a secret is done using Open Service Broker domain, for example a plan. KIM is not aware of a plan.

Reply:
So indeed KEB knows the secret and will inject it? ok